hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -1660,7 +1660,10 @@ public static String deleteMetadataTablePartition(HoodieTableMetaClient dataMeta
    * @param index         Index of the file group within the partition
    * @return The fileID
    */
-  public static String getFileIDForFileGroup(MetadataPartitionType partitionType, int index) {
+  public static String getFileIDForFileGroup(MetadataPartitionType partitionType, int index, String partitionName) {
```

Comment:
wrt to my comment 
https://github.com/apache/hudi/pull/12259/files#r1842787725 

I am asking about, why replace "_" with "-". 
I was checking the parsing logic for getFileIDForFileGroup(), getFileGroupIndexFromFileId, and I don't think its strictly requires to replace "_". 
so, was just curious.